master/internal/db/postgres_rp_workspace_bindings.go

```diff
@@ -170,3 +168,91 @@ func OverwriteRPWorkspaceBindings(ctx context.Context,
 	err = AddRPWorkspaceBindings(ctx, workspaceIds, poolName, resourcePools)
 	return err
 }
+
+// GetUnboundRPs get unbound resource pools.
+func GetUnboundRPs(
+	ctx context.Context, resourcePools []config.ResourcePoolConfig,
+) ([]string, error) {
+	var boundResourcePools []string
+	_, err := Bun().NewSelect().
+		Column("pool_name").
+		Table("rp_workspace_bindings").
+		Distinct().
+		Exec(ctx, &boundResourcePools)
+	if err != nil {
+		return nil, err
+	}
+
+	boundRPsMap := map[string]bool{}
+	for _, boundRP := range boundResourcePools {
+		boundRPsMap[boundRP] = true
+	}
+
+	unboundRPs := []string{}
+	for _, resourcePool := range resourcePools {
+		if !boundRPsMap[resourcePool.PoolName] {
+			unboundRPs = append(unboundRPs, resourcePool.PoolName)
+		}
+	}
+
+	return unboundRPs, nil
+}
+
+// RP is a helper strct for Bun query.
+type RP struct {
+	Name string
+}
+
+// ReadRPsAvailableToWorkspace returns the names of resource pool bound to a
+// workspace.
+func ReadRPsAvailableToWorkspace(
+	ctx context.Context,
+	workspaceID int32,
+	offset int32,
+	limit int32,
+	resourcePoolConfig []config.ResourcePoolConfig,
+) ([]string, *apiv1.Pagination, error) {
+	unboundRPNames, err := GetUnboundRPs(ctx, resourcePoolConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	unboundRPs := []*RP{}
+	for _, unboundRPName := range unboundRPNames {
+		unboundRPs = append(unboundRPs, &RP{unboundRPName})
+	}
+
+	var rpNames []string
+	var query *bun.SelectQuery
+	if len(unboundRPs) > 0 {
+		values := Bun().NewValues(&unboundRPs)
```

Comment:
interesting, do we need to pass in a struct of just a string for this to work?

Reply:
I wish there's an cleaner/easier way.

I went with this way based on Bun VALUES [example](https://bun.uptrace.dev/guide/query-common-table-expressions.html#values) and the [doc](https://pkg.go.dev/github.com/uptrace/bun#DB.NewValues):

> ```
> values := db.NewValues(&[]*Book{book1, book2})
> 
> res, err := db.NewUpdate().
>     With("_data", values).
>     Model((*Book)(nil)).
>     Table("_data").
>     Set("title = _data.title").
>     Set("text = _data.text").
>     Where("book.id = _data.id").
>     Exec(ctx)
> ```

> ```
> func (db *DB) NewValues(model interface{}) *ValuesQuery
> ```

And according to [Defining models](https://bun.uptrace.dev/guide/models.html#mapping-tables-to-structs):

> For each table you need to define a corresponding Go struct (model). Bun maps the exported struct fields to the table columns and ignores the unexported fields.

I found [a way to store output of a query](https://github.com/determined-ai/determined/pull/7176/files#diff-5b415542f8c43305031c56d94628f15cd0152c001c82988baad2d7b485e1b3d8R249) into a slice of string instead of a slice of model. But so far, I haven't found a way to to the same in NewValues().

> ```
> query.Scan(ctx, &rpNames)
> ```

Reply:
ok we can ship this first and then investigate later if we can simplify it. Can you make a TODO comment?

Reply:
Added